src/main/java/eu/yals/services/LinkService.java

```diff
@@ -1,16 +1,58 @@
 package eu.yals.services;
 
+import eu.yals.models.Link;
+import eu.yals.models.dao.LinkRepo;
 import eu.yals.result.GetResult;
 import eu.yals.result.StoreResult;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataAccessResourceFailureException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.CannotCreateTransactionException;
+
+import java.util.Optional;
 
 /**
- * Queries for Link storage
+ * Service, which interacts with database to store/retrieve links.
  *
  * @since 2.0
  */
-public interface LinkService {
+@Slf4j
+@Service
+public class LinkService {
+
+    private final LinkRepo repo;
+
+    public LinkService(LinkRepo repo) {
+        this.repo = repo;
+    }
+
+    public GetResult getLink(String ident) {
+        Optional<Link> result;
+        try {
+            result = repo.findSingleByIdent(ident);
+        } catch (Exception e) {
+            if (e instanceof DataAccessResourceFailureException) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [An instanceof check is being performed on the caught exception.  Create a separate catch clause for this exception type.](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5392475)